examples/seismic/model.py

```diff
@@ -303,37 +303,33 @@ class Model(object):
     """
     def __init__(self, origin, spacing, shape, vp, nbpml=20, dtype=np.float32,
                  epsilon=None, delta=None, theta=None, phi=None):
-        self.origin = origin
-        self.spacing = spacing
         self.shape = shape
         self.nbpml = int(nbpml)
-        self.dtype = dtype
 
-        # Ensure same dimensions on all inpute parameters
-        assert(len(origin) == len(spacing))
-        assert(len(origin) == len(shape))
+        shape_pml = np.array(shape) + 2 * self.nbpml
+        extent = tuple(np.array(spacing) * (shape_pml))
```

Comment:
`tuple(origin + np.array(spacing) * (shape_pml -1)`

Reply:
Again, different interpretations of `extent`.

Reply:
Uhhh, and I tried the `- 1`, but things started to break. So we might need to go through the model code carefully to make sure that everything treats nodes and cells accordingly.

Reply:
Even with the definition of extent, domain starts at `origin` and finishes at `origin+extent` in n points, so that's (n-1) intervals.

Reply:
Ahhh yes, you are right. I know what I did wrong previously. Will fix.